lib/ContextChat/ContextChatProvider.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Mail\ContextChat;
+
+use OCA\Mail\AppInfo\Application;
+use OCA\Mail\Db\Message;
+use OCA\Mail\Db\MessageMapper;
+use OCA\Mail\Events\MessageDeletedEvent;
+use OCA\Mail\Events\NewMessagesSynchronized;
+use OCA\Mail\Service\AccountService;
+use OCA\Mail\Service\ContextChat\TaskService;
+use OCA\Mail\Service\MailManager;
+use OCP\BackgroundJob\IJobList;
+use OCP\ContextChat\Events\ContentProviderRegisterEvent;
+use OCP\ContextChat\IContentManager;
+use OCP\ContextChat\IContentProvider;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\IURLGenerator;
+use OCP\IUser;
+use OCP\IUserManager;
+
+/**
+ * @implements IEventListener<Event>
+ */
+class ContextChatProvider implements IContentProvider, IEventListener {
+
+	public const CONTEXT_CHAT_MESSAGE_MAX_AGE = 60 * 60 * 24 * 365;
```

Comment:
The .25 is to account for leap years.

Also, the comments should be added back for clarity.